pkgs/build-support/rust/build-rust-package/default.nix

```diff
@@ -67,7 +67,7 @@ let
 
   cargoDeps =
     if cargoVendorDir != null then null
-    else if cargoLock != null then importCargoLock cargoLock
+    else if cargoLock != null then importCargoLock ({ patches = cargoPatches; } // cargoLock)
```

Comment:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/wayland/aw-watcher-window-wayland/default.nix uses both `cargoLock` and `cargoPatches`  so I had to remove the automatic passing of patches.